src/components/algolia/algolia-course-related.tsx

```diff
@@ -0,0 +1,84 @@
+"use client"
+
+import {useMemo} from "react"
+import algoliasearch from "algoliasearch/lite"
+import {RelatedProducts} from "react-instantsearch"
+import {InstantSearchNext} from "react-instantsearch-nextjs"
+import ImageCard from "@components/patterns/image-card"
+import {H3} from "@components/elements/headers"
+import ActionLink from "@components/elements/action-link"
+
+type Props = {
+  objectID: string
+  appId: string
+  searchIndex: string
+  searchApiKey: string
+}
+
+type RelatedCourseItemProps = {
+  sum_course_catalog_number?: string
+  sum_course_class_number?: string
+  title?: string
+  photo?: string
+  url?: string
+}
+
+const RelatedCourseItem = ({item}: {item: RelatedCourseItemProps}) => {
+  return (
+    <ImageCard
+      imageUrl={item.photo}
+      imageAlt=""
+    >
+      <div className="flex flex-col">
+        <H3 className="rs-mb-0 order-2 max-w-[900px] font-roboto font-normal">{item.title}</H3>
+        <div className="rs-mb-0 order-1 font-normal">{item.sum_course_catalog_number}</div>
+        {item.url && (
+          <ActionLink
```

Comment:
Use an `aria-labelledby` prop on the link and target the H3 header so we don't have multiple links with the same text.